### PR TITLE
Update azurerm to latest version 2.39

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/11474

### CODE REVIEW

File: modules/azure/blob/blob.tf [hunk 1]
```diff
@@ -1,16 +1,15 @@
 resource "azurerm_storage_account" "storage_acc" {
   name                      = "${var.cluster_name}config"
-  resource_group_name       = var.resource_group_name
   location                  = var.azure_location
   account_kind              = "BlobStorage"
   account_tier              = "Standard"
   account_replication_type  = "LRS"
   enable_https_traffic_only = true
+  resource_group_name   = var.resource_group_name
```
Comment (whites11) [inline modules/azure/blob/blob.tf:8]:
```suggestion
  resource_group_name       = var.resource_group_name
```

**Review by whites11 — APPROVED** [review]
LGTM when tested

**Comment by corest** [comment]
Works locally, but fails in github action. Still need to debug

**Comment by whites11** [comment]
FYI this was a blocker for bumping azurerm terraform in the past and apparently still is: https://github.com/terraform-providers/terraform-provider-azurerm/issues/2489#issuecomment-742027523

**Comment by corest** [comment]
We don't have more subnets. And we need to switch to vmss and for that we need new provider version

**Comment by sanderginn** [comment]
Hi @whites11 , I see you referenced my comment. I'd like to let you know that I had overlooked a resource and the proposed fix in the thread does seem to do the job.

**Comment by whites11** [comment]
> Hi @whites11 , I see you referenced my comment. I'd like to let you know that I had overlooked a resource and the proposed fix in the thread does seem to do the job.

Thanks for letting me know @sanderginn

**Pull request merged**